snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -84,6 +84,7 @@ def __init__(
         self.part_dir = os.path.join(project.parts_dir, self.name)
         self.part_source_dir = os.path.join(self.part_dir, "src")
         self.part_build_dir = os.path.join(self.part_dir, "build")
+        self.part_build_plugin_dir = os.path.join(self.part_dir, "build_plugin")
```

Comment:
This directory needs to be removed in the `clean_build()` method defined in this class.

Reply:
Done in 24f8a494b7d9b3948c04a1684c1a67420715b242.